### ISSUE TITLE
6LowPan outgoing fragmentation error

### ISSUE THREAD

**Issue by benbrittain** [smoltcp-rs/smoltcp#640]
I have an echo server running on a microcontroller, it successfully reads in fragmented 6LowPan packets and assembles them, but when I turn around to blast them out again, I get what looks like a never ending stream of spurious fragments?

Wireshark shows the actual message contents in the first two packets (#22 & #23), then a repeating write of packets of length 26.

![image](https://user-images.githubusercontent.com/207432/179264640-0039faa2-bbca-4e9a-abd7-5f844ed37981.jpg)

**Comment by benbrittain** [comment]
The phy device is operating as a null sink during the time we are sending the outgoing fragments, always consuming the packet to help isolate the problem. I've attached a wireshark txt export so you can see the actual packets.

The Bad FCS errors can be ignored, on the wire they have the right checksum since it's handled by the radio peripheral I'm using.

[raw-out.txt](https://github.com/smoltcp-rs/smoltcp/files/9122437/out.txt)

Let me know if I can provide any more context @thvdveld !

**Comment by thvdveld** [comment]
Could you maybe share the pcap file? Then I can open it with Wireshark as well.

**Comment by benbrittain** [comment]
Github won't let me upload pcap files, so I've just renamed it as .txt and I hope they don't mangle it. 
[out.txt](https://github.com/smoltcp-rs/smoltcp/files/9122539/out.txt)

**Comment by thvdveld** [comment]
Something is wrong with the calculation of the first fragment packet size and the n fragment packet size (I guess). I'll try to have a look at it this weekend, otherwise it will be in two weeks (holidays).

**Comment by benbrittain** [comment]
Obviously no rush. I think you are right, I'm actually in the middle of printf debugging the `dispatch_ieee802154_out_packet` function and something definitely seems wonky with the fragn size calculation.

**Comment by benbrittain** [comment]
poked at it a little this morning. Something about the fragn calculation seems off to me.

```
let frag_size = (*packet_len - *sent_bytes).min(*fragn_size);
```

this evaluates to 0, after the second packets (as expected!), but that means nothing gets copied into the buffer since we've you know, sent all the relevant data. We still get packets being generated as well.

**Comment by thvdveld** [comment]
I think we can close this now, @benbrittain ? (fixed by #645)
